src/main/java/com/easypost/easyvcr/internalutilities/ApachePatch.java

```diff
@@ -0,0 +1,279 @@
+package com.easypost.easyvcr.internalutilities;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.BitSet;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * This class is a collection of Apache functions that have been extracted for use in this application.
+ * This class is required because Apache's libraries are notorious for security vulnerabilities, so we'll only
```

Comment:
I'd like future-proof links to where we pulled these from for reference and future travelers please.

Also yanking code directly from another project may not be appropriate depending on licensing (another reason for links).